### ISSUE TITLE
search for one disease get responses for others without explanation

### ISSUE THREAD

**Issue by TranslatorIssueCreator** [NCATSTranslator/Feedback#632]
## Steps to reproduce:
MVP1 on Hereditary Sensory And Autonomic Neuropathy

**Comment by sandrine-muller-research** [comment]
The first result display erythomelalgia instead of the queried disease:
<img width="962" alt="image" src="https://github.com/NCATSTranslator/Feedback/assets/22616936/d95a2379-9785-4283-bc35-3949c6a51373">
from ARAX Translator Reasoner.

**Comment by amykglen** [comment]
hi @sandrine-muller-research - in the ARAX response we indicate that `Erythromelalgia` is a subclass of `Hereditary Sensory And Autonomic Neuropathy` in the result support graphs: 

![Screen Shot 2023-12-11 at 10 41 09 AM](https://github.com/NCATSTranslator/Feedback/assets/49423686/0e15196d-9fa7-4dad-8220-9c71b05f2a1b)

(screenshot is from the [ARAX response](https://arax.ncats.io/beta/?r=176030) for that PK)

in our understanding this is an acceptable way of handling subclass information, since no standard method of capturing such info in TRAPI has yet been agreed upon. if we remapped the `Erythromelalgia` edges in the main result graph to point to `Hereditary Sensory And Autonomic Neuropathy` instead, we wouldn't be able to communicate which treats edges actually went through `Erythromelalgia`, rather than the parent concept. perhaps the Translator UI could display the subclass information that's captured in the support graph?

**Comment by cbizon** [comment]
@amykglen can you say where that subclass edge is coming from?  Erythromelagia is not a subclass of hereditary sensory and autonomic neuropathy in MONDO as far as I can tell

**Comment by cbizon** [comment]
Yeah, I'm not finding it in robokop either.  @vdancik could this be coming out of molepro?

**Comment by amykglen** [comment]
notably, upon rerunning the query (previously I was looking at the response from back in November), it looks like Molepro is responsible for all of the 'Erythromelagia' results:

https://arax.ncats.io/beta/?r=237926

![Screenshot 2024-05-01 at 10 37 18 AM](https://github.com/NCATSTranslator/Feedback/assets/49423686/eb544b64-edd5-48d8-a6b0-40662c29690f)

**Comment by sandrine-muller-research** [comment]
Please note this is a test of the UI, not of the ARAX UI. I reran the query, [here](https://ui.test.transltr.io/main/results?l=Hereditary%20Sensory%20And%20Autonomic%20Neuropathy&i=MONDO:0015364&t=0&r=0&q=9b4d4f6f-954d-4904-bef9-1c2aeb190a4c) is the link.
The issue is not the knowledge about _Erythromelagia_, the issue is that the user was asking a question about _Hereditary Sensory And Autonomic Neuropathy_ and we are getting results about _Erythromelagia_ without having the explanation in the subgraph shown (in the UI).  
Is MolePro returning _Erythromelagia_ when you query it for _Hereditary Sensory And Autonomic Neuropathy_ or when you query it for _Erythromelagia_?

From the rerun query, the user can see:
![image](https://github.com/NCATSTranslator/Feedback/assets/22616936/a432093d-aaae-4f8b-a7b0-0fecf7d0c938)
As a user I would have expected the first path to be:
Aspirin --treats--> Erythromelalgia --subclass_of--> Hereditary Sensory And Autonomic Neuropathy

That is the issue I was trying to describe here.

**Comment by vdancik** [comment]
> Is MolePro returning Erythromelagia when you query it for Hereditary Sensory And Autonomic Neuropathy or when you query it for Erythromelagia?

Indeed MolePro returns Erythromelagia since it is a subclass of Hereditary Sensory And Autonomic Neuropathy according to https://automat.transltr.io/ubergraph/1.4/query

```
{
    "message": {
        "query_graph": {
            "edges": {
                "e00": {
                    "subject": "n01",
                    "object": "n00",
                    "predicates": [
                        "biolink:subclass_of"
                    ]
                }
            },
            "nodes": {
                "n00": {
                    "ids": [
                        "MONDO:0015364"
                    ],
                    "categories": [
                        "biolink:Disease"
                    ]
                },
                "n01": {
                    "categories": [
                        "biolink:Disease"
                    ]
                }
            }
        }
    }
}
```

**Comment by cbizon** [comment]
True in ubergraph PROD and TEST, but apparently no longer the case in DEV or CI. @vdancik does molepro need to be updated with the new ubergraph or is that supposed to happen at query time?

**Comment by vdancik** [comment]
@cbizon , MolePro obtains subclass information at build time, so to get this fixed we will have to wait for a next build.

**Comment by cbizon** [comment]
OK - does that happen on a schedule?

**Comment by vdancik** [comment]
We are targeting quarterly builds, the reality is more like every four months. Last build was Apr 1st, previous Dec 15.

**Comment by sierra-moxon** [comment]
@vdancik - is there a new build of MolePro in the CI environment coming this month?  Or is it more like September?

**Comment by codewarrior2000** [comment]
@sierra-moxon Vlado is out this week but just to get an answer to you by today, MolePro has a Fugu release already in CI.
See changelog [here](https://docs.google.com/spreadsheets/d/1v3vdhlrZJIOFP1FYVp_9FmTBn6dTcZfYsHpUx5R10JE/edit?gid=1382644306#gid=1382644306).
